### ISSUE TITLE
"Skipping rendezvous re-initialization" when calling collective_ops in a distributed cluster

### ISSUE THREAD

**Issue by whhu** [tensorflow/tensorflow#33321]
We meet a [problem](https://stackoverflow.com/questions/58161090/when-does-skipping-rendezvous-re-initialization-emerge-in-calling-collective) which continuously logs `re-initialization` within the same MonitoredSession on the leader worker. Furthermore, there seems to be an accuracy losing during the gradient descent of the parabolic toy model. We have no idea how to prevent the logging of `re-initialization`. All hints or explanations are welcome.

It seems the [`BaseRemoteRendezvous::Initialize`](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/core/distributed_runtime/base_rendezvous_mgr.cc#L152) method is called at every train step. The log of the leader worker is attached below.
```console
task 0: before syncing ...
{'u': 0.1, 'loss': -0.19, 'global_step': 0}
task 0: running init_ops ...
{'u': 0.1, 'loss': -0.19, 'global_step': 0}
task 0: obtaining gstep ...
train one step at step 0.
2019-09-30 10:15:56.954439: I tensorflow/core/distributed_runtime/base_rendezvous_mgr.cc:159] Skipping rendezvous re-initialization.
{'u': 0.1, 'loss': -0.19, 'global_step': 1}
train one step at step 1.
2019-09-30 10:15:56.959006: I tensorflow/core/distributed_runtime/base_rendezvous_mgr.cc:159] Skipping rendezvous re-initialization.
{'u': 0.28, 'loss': -0.4816, 'global_step': 2}
train one step at step 2.
2019-09-30 10:15:56.962205: I tensorflow/core/distributed_runtime/base_rendezvous_mgr.cc:159] Skipping rendezvous re-initialization.
{'u': 0.42396685, 'loss': -0.66818583, 'global_step': 3}
train one step at step 3.
2019-09-30 10:15:56.965719: I tensorflow/core/distributed_runtime/base_rendezvous_mgr.cc:159] Skipping rendezvous re-initialization.
{'u': 0.53912044, 'loss': -0.78759, 'global_step': 4}
train one step at step 4.
2019-09-30 10:15:56.969335: I tensorflow/core/distributed_runtime/base_rendezvous_mgr.cc:159] Skipping rendezvous re-initialization.
{'u': 0.6312327, 'loss': -0.8640107, 'global_step': 5}
{'u': 0.70491827, 'loss': -0.9129268, 'global_step': 5}
task 0: final syncing ...
task 0: finalized
```
A related issue #32810 also has the similar `re-initialization` at first. But the logging disappears with the improved code snippet, which patches the `send_tensor`, `recv_tensor` and `ChiefSessionCreator` on both workers.

**Comment by ravikyram** [comment]
@whhu 
I believe the code provided in Stack overflow is the reproducible code with TF version 1.13. Please, confirm.Thanks!

**Comment by whhu** [comment]
I'm sorry the code is not the reproducible one. We come across the `re-initialization` problem in a private project, which makes use of the `collective_ops` in a similar way as the stackoverflow [question](https://stackoverflow.com/questions/58161090/when-does-skipping-rendezvous-re-initialization-emerge-in-calling-collective). This weird problem occurs only in our private project. That is why we want to find out when TensorFlow logs `Skipping rendezvous re-initialization`.

We try [`broadcast_send`](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/ops/collective_ops.py#L61)/[`broadcast_recv`](https://github.com/tensorflow/tensorflow/blob/r1.13/tensorflow/python/ops/collective_ops.py#L108) and find out that two workers hang and the logging is triggered by worker 0 when worker 1 fetches the collective tensor defined on worker 0. I wish this maybe helpful to illustrate the problem.

**Comment by whhu** [comment]
The evidence looks like irrelevant to the `collective_ops`. In our project, the accessing to global_step triggers the `re-initialization` problem, which is hard to imagine. (-\_-)!!
Many thanks for the response all the same. I will close this issue and may start a new one when we locate the problem.
